BDCSCli/Commands.hs

```diff
@@ -57,6 +64,35 @@ humanRecipesList :: Response Value -> String
 humanRecipesList jsonValue = intercalate ", " $ map T.unpack recipes
   where recipes = jsonValue ^.. responseBody . key "recipes" . values . _String
 
+-- | Process the compose types command
+-- Prints a list of the supported compose types
+composeCommand :: Session -> CliOptions -> [String] -> IO ()
+composeCommand _ _ ("types":_) =
+    -- API has a list of compose types, but we are not currently using that
+    putStrLn "tar"
+
+composeCommand _    _    ("tar":[])       = putStrLn "ERROR: Missing recipe name"
+composeCommand sess opts ("tar":recipe:_) = do
+    r <- depsolveRecipes sess opts recipe
+    when (isJust r) $ do
+        let deps = decodeDepsolve $ fromJust r
+--        when (isJust deps) $ composeTar opts $ fromJust deps
+        forM_ deps composeTar
+  where
+    composeTar deps = withSystemTempFile "bdcs-deps-" $ \tmpFile hFile -> do
+        -- write deps to tmpFile
+        hPutStr hFile $ intercalate "\n" $ getDepNEVRAList deps
+        hClose hFile
+        let tarFile = tarFileName recipe
+        let mddbPath = optMDDB opts
+        let repoPath = optRepo opts
+        printf "Creating tar of %s, saving to %s\n" recipe tarFile
+        rawSystem "export" [mddbPath, repoPath, tarFile, tmpFile]
+        removeFile tmpFile
```

Comment:
You shouldn't need to remove the temp file at the end.  withSystemTempFile should do that for you.